frontend/src/metabase/query_builder/components/filters/modals/BulkFilterModal/BulkFilterModal.tsx

```diff
@@ -44,6 +55,12 @@ const BulkFilterModal = ({
     return query.topLevelFilterFieldOptionSections(null, 2, true);
   }, [query]);
 
+  const searchItems = useDebouncedFunction(
+    () => getSearchHits(searchQuery, sections),
+    200,
+    [searchQuery, sections],
+  );
```

Comment:
there's some pretty significant processing overhead involved in filtering all these dimensions, so we need to debounce this.